packages/core/test/parse.spec.ts

```diff
@@ -46,6 +47,32 @@ describe('parse', () => {
     expect(replacementFn).toHaveBeenCalledWith(query, undefined);
   });
 
+  it('Should not yield app context during onParse', async () => {
+    const favoritePetSpy = jest.fn((_args: any) => {});
+    interface AppContext {
+      favoritePet: string;
+    }
+    const useFavoritePet: Plugin<AppContext> = {
+      onContextBuilding({ extendContext }) {
+        extendContext({
+          favoritePet: 'turtle',
+        });
+      },
+    };
+    const useOnParse: Plugin<AppContext> = {
+      onParse: () => {
+        return ({ context }) => {
+          // @ts-expect-error favoritePet is not available during onParse
```

Comment:
This is the "failing spec", as you can see on [L73](https://github.com/dotansimha/envelop/pull/1347/files#diff-63670871751ebf486fd2051bc6eeb7ffa5001577f118331bd008a3a2fe0b801fR73) this is undefined but TS told me it would be there